pkg/controller/controller.go

```diff
@@ -395,10 +397,10 @@ func (c *Controller) bootWithError(ctx context.Context) error {
 				Reconciler:              c,
 			}).
 			WithEventFilter(predicate.Funcs{
-				CreateFunc:  func(e event.CreateEvent) bool { return c.selector.Matches(selector.NewLabels(e.Meta.GetLabels())) },
-				DeleteFunc:  func(e event.DeleteEvent) bool { return c.selector.Matches(selector.NewLabels(e.Meta.GetLabels())) },
-				UpdateFunc:  func(e event.UpdateEvent) bool { return c.selector.Matches(selector.NewLabels(e.MetaNew.GetLabels())) },
-				GenericFunc: func(e event.GenericEvent) bool { return c.selector.Matches(selector.NewLabels(e.Meta.GetLabels())) },
+				CreateFunc:  func(e event.CreateEvent) bool { return c.selector.Matches(labels.Set(e.Meta.GetLabels())) },
+				DeleteFunc:  func(e event.DeleteEvent) bool { return c.selector.Matches(labels.Set(e.Meta.GetLabels())) },
+				UpdateFunc:  func(e event.UpdateEvent) bool { return c.selector.Matches(labels.Set(e.MetaNew.GetLabels())) },
+				GenericFunc: func(e event.GenericEvent) bool { return c.selector.Matches(labels.Set(e.Meta.GetLabels())) },
```

Comment:
This doesn't look like server-side filtering to me. Am I missing something?

Reply:
I believe [this](https://github.com/giantswarm/operatorkit/blob/047b5dd3303e00d6d0c772121ae8841bea3d1ee4/pkg/controller/collector/timestamp.go#L79) is the line that does the magic.

Reply:
Oh yeah, the part about the server-side filtering is only relevant to the timestamp collector linked by Yulia. This `WithEventFilter` part was a secondary change related to refactoring the selectors/labels to keep everything consistent. It shouldn't have any effect on watches.

Reply:
I took me a while to check this. When initially implementing this part I designed the Selector interface in a way that k8s.io/apimachinery/pkg/labels.Selector fulfils that interface (I extracted only the Matches method) but I also wanted to be allow writing a simple struct that fulfils this interface. So you can pass it without making breaking changes. So I believe you can drop all changes to `controller.go` and your tests will still compile and be green.

Reply:
Interesting. I have trouble with go interfaces sometimes. I'll give it a try.

Reply:
The problem is that we pass this selector into [ListOptions's LabelSelector which accepts an object satisfying k8s.io/apimachinery/pkg/labels.Selector interface](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.8.2/pkg/client/options.go#L321-L323) and the `Selector` from `operatorkit` doesn't define all of the required methods so we need to use `Selector` from `apimachinery` throughout (my approach so far which requires more work for consumers) or make the `Selector` from `operatorkit` define the same interface as the one from `apimachinery` (less work for consumers, more work in `operatorkit`). What do you think @kopiczko?

Reply:
I see. Then it doesn't work. I leave the decision for you.